packages/react-core/src/components/DataList/DataList.tsx

```diff
@@ -30,6 +42,8 @@ export interface DataListProps extends Omit<React.HTMLProps<HTMLUListElement>, '
   selectedDataListItemId?: string;
   /** Flag indicating if DataList should have compact styling */
   isCompact?: boolean;
+  /** Specifies the grid breakpoints  */
+  gridBreakPoint?: 'grid-none' | 'grid' | 'grid-sm' | 'grid-md' | 'grid-lg' | 'grid-xl' | 'grid-2xl';
```

Comment:
Change to:
```tsx
/** Modifies the data list to switch to a grid layout at a specified breakpoint */
gridBreakpoint?: 'none' | 'always' | 'sm' | 'md' | 'lg' | 'xl' | '2xl';
```

You can support `none | always` through more string transforms, or just declare this at the top of the file:
```ts
const gridBreakpointClasses = {
  'none': 'pf-m-grid', // Placeholder per https://github.com/patternfly/patternfly-react/issues/4965#issuecomment-704984236
  'always': styles.modifiers.grid,
  'sm': styles.modifiers.gridSm,
  'md': styles.modifiers.gridMd,
  'lg': styles.modifiers.gridLg,
  'xl': styles.modifiers.gridXl,
  '2xl': styles.modifiers.grid_2xl,
};
```
and then use `gridBreakPoint && gridBreakpointClasses[gridBreakPoint]`.